src/main/java/bio/overture/songsearch/model/Run.java

```diff
@@ -38,12 +40,22 @@
 
   private String runId;
 
-  private List<Analysis> producedAnalyses;
+  private DataFetcher<List<Analysis>> producedAnalysesFetcher;
 
-  private List<Analysis> inputAnalyses;
+  private DataFetcher<List<Analysis>> inputAnalysesFetcher;
 
   @SneakyThrows
   public static Run parse(@NonNull Map<String, Object> sourceMap) {
     return MAPPER.convertValue(sourceMap, Run.class);
   }
+
+  @SneakyThrows
+  public List<Analysis> getInputAnalyses(DataFetchingEnvironment env) {
+    return inputAnalysesFetcher.get(env);
+  }
+
+  @SneakyThrows
+  public List<Analysis> getProducedAnalyses(DataFetchingEnvironment env) {
```

Comment:
this is how graphql-java provides the variables for a nested field: https://www.graphql-java.com/documentation/v14/data-fetching/